pkg/proxy/utils.go

```diff
@@ -2,23 +2,68 @@ package proxy
 
 import (
 	"archive/tar"
+	"archive/zip"
 	"bytes"
 	"compress/gzip"
 	"context"
+	"fmt"
+	"io"
 	"strconv"
 
-	windtunnelv1alpha1 "github.com/CarnegieMellon-PlantD/PlantD-operator/api/v1alpha1"
+	plantdv1alpha1 "github.com/CarnegieMellon-PlantD/PlantD-operator/api/v1alpha1"
 	"github.com/CarnegieMellon-PlantD/PlantD-operator/pkg/datagen"
 	"github.com/CarnegieMellon-PlantD/PlantD-operator/pkg/errors"
-	"github.com/CarnegieMellon-PlantD/PlantD-operator/pkg/utils"
 
 	"github.com/brianvoe/gofakeit/v6"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
 )
 
-// GetDataset retrieves a DataSet object by namespace and name.
-func GetDataset(ctx context.Context, c client.Client, namespace string, name string) (*windtunnelv1alpha1.DataSet, error) {
-	dataset := &windtunnelv1alpha1.DataSet{}
+// Constants defining the possible values for GVK (Group, Version, Kind).
+// All values can be used in the schema.GroupVersionKind struct.
+const (
+	PlantDGroup      string = "windtunnel.plantd.org"
+	V1Alpha1Version  string = "v1alpha1"
```

Comment:
Duplicated. The original one is under `api/v1alpha1/doc.go`.

Reply:
Fixed. Now it's using the values in `api/v1alpha1/doc.go`.